### PR TITLE
Added onErrorReturn, onErrorResumeNext and onErrorComplete operators

### PR DESCRIPTION
**Pull request by dmitryustimov**

### CODE REVIEW

File: reaktive/src/commonMain/kotlin/com/badoo/reaktive/completable/OnErrorResumeNext.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+        subscribeSafe(
+            object : CompletableObserver, CompleteCallback by observer {
+
+                override fun onSubscribe(disposable: Disposable) {
+                    disposableWrapper.set(disposable)
+                }
+
+                override fun onError(error: Throwable) {
+                    observer.tryCatch({ nextSupplier(error) }) {
```
Comment (arkivanov) [inline reaktive/src/commonMain/kotlin/com/badoo/reaktive/completable/OnErrorResumeNext.kt]: If `nextSupplier` throws an exception then there are two exceptions: `error` and the new one. You should compose them into `CompositeException`. Check out `doOnBeforeError` operator for usage example. Same for other similar places. Also please fix this for `Observable.onErrorResumeNext`, looks like I missed when I was reviewing that.

**Comment by arkivanov** [comment]
We are going to release `1.0.0`, let's have it there :+1:

File: reaktive/src/commonMain/kotlin/com/badoo/reaktive/completable/OnErrorResumeNext.kt (hunk 1 above)
Comment (dmitryustimov) [reply reaktive/src/commonMain/kotlin/com/badoo/reaktive/completable/OnErrorResumeNext.kt]: Thanks for the reference to `doOnBeforeError`. Now both exceptions will be wrapped into `CompositeException` when supplier call fails.
Also please take a look at `TestObserver.assertError` with predicate

File: reaktive/src/commonTest/kotlin/com/badoo/reaktive/maybe/OnErrorResumeNextTest.kt [hunk 2]
```diff
@@ -119,11 +120,16 @@ class OnErrorResumeNextTest :
...
-
-        observer.assertError(throwable)
+        val upstreamThrowable = Throwable()
+        val supplierThrowable = Throwable()
+        val observer = upstream.onErrorResumeNext { throw supplierThrowable }.test()
+        upstream.onError(upstreamThrowable)
+
+        observer.assertError { throwable ->
+            throwable is CompositeException &&
+                    throwable.cause1 == upstreamThrowable &&
```
Comment (arkivanov) [inline reaktive/src/commonTest/kotlin/com/badoo/reaktive/maybe/OnErrorResumeNextTest.kt]: We are following Kotlin code styles, continuation indent should be 4 spaces, not 8.
We will consider to fix it in the repo somehow, but please change it manually in IDEA settings and reformat all the code.
image

Comment (dmitryustimov) [reply reaktive/src/commonTest/kotlin/com/badoo/reaktive/maybe/OnErrorResumeNextTest.kt]: Fixed

**Pull request merged**